js/heists-filtering.ts

```diff
@@ -0,0 +1,196 @@
+interface Challenge {
+    id: number;
+    name: string;
+    category: string;
+    score?: number;
+    description: string;
+    isBeta?: boolean;
+}
+
+class HeistsFilter {
+    private challenges: Challenge[] = [];
+    private currentFilter: string = 'all';
+    private searchTerm: string = '';
+
+    constructor() {
+        this.initializeChallenges();
+        this.setupEventListeners();
+    }
+
+    private initializeChallenges(): void {
+        // Get all challenge cards from the DOM
+        const challengeCards = document.querySelectorAll('[data-challenge-card]');
+        
+        challengeCards.forEach(card => {
+            const challenge: Challenge = {
+                id: parseInt(card.getAttribute('data-challenge-id') || '0'),
+                name: card.querySelector('[data-challenge-name]')?.textContent?.trim() || '',
+                category: card.getAttribute('data-challenge-category') || '',
+                score: this.parseScore(card.querySelector('[data-challenge-score]')?.textContent),
+                description: card.querySelector('[data-challenge-description]')?.textContent?.trim() || '',
+                isBeta: card.hasAttribute('data-challenge-beta')
+            };
+            this.challenges.push(challenge);
+        });
+    }
+
+    private parseScore(scoreElement: Element | null): number | undefined {
```

Comment:
Maybe set the score in a data attribute directly rather than searching for it in the textContent with regex, like you do with id and name